wordpress-mobile/WordPress-iOS
# Move Core Data updates to background in Editor Settings

Related to #21008.

There are two main changes in this PR:
1. Removing the `performQuery` call that returns `WPAccount`.
2. Using `performAndSave` in the `updateAllSites` to ensure it runs in the right context queue.

## Test Instructions

The existing unit test should cover the changes in `updateAllSites`.

Resuming the app from background is the only path to call the `updateAllSites` function. First, we'll need to make some code changes: [`GutenbergRollout.shouldPerformPhase2Migration`](https://github.com/wordpress-mobile/WordPress-iOS/blob/27bf04400b0382a1ec383d4f551996b50be4ff10/WordPress/Classes/Utility/Editor/GutenbergRollout.swift#L20) needs to return `true`. After the code change, launch the app from Xcode, make the app go to background, and bring it back to foreground. The `updateAllSites` should be executed when [the app resumes from background](https://github.com/wordpress-mobile/WordPress-iOS/blob/27bf04400b0382a1ec383d4f551996b50be4ff10/WordPress/Classes/System/WordPressAppDelegate.swift#L211). As a result, all your blogs should now has Gutenberg editor enabled.

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
What's described in the Test Instructions section

3. What automated tests I added (or what prevented me from doing so)
None.

PR submission checklist:


UI Changes testing checklist: N/A
